### DISCUSSION TITLE
Title scale

### DISCUSSION THREAD

**Discussion by AnaMontes11** [Telefonica/mistica-design#1125] [Proposals]
A veces se usa el título del Header como solución a casos en los que se requiere de un título principal para abrir página.

¿Tiene sentido crear una escala de "titles" más completa, que evite este tipo de usos?

Podríamos establecer una escala de titles acotada por jerarquías de uso tipo:

<img width="486" alt="titles" src="https://user-images.githubusercontent.com/88187691/224999437-c09b0497-ffe7-4d5e-af94-8dd5b9f78cfe.png">

**Comment by yceballost** [comment]
¿Cómo encaja esto en mobile con una navegación nativa?
Debería existir ese page title cuando ya existe en la navbar?
Si no existe, en el body empezaría uasndo ese "section title"? (no le metería nombres tan semánticos porque luego pueden ser usados en otros sitios, soy más fan de lo agnóstico y en la documentación poner ejemplos de usos)

en brand factory hay una tabla que se hizo al principio de definir textos
https://brandfactory.telefonica.com/document/1846#/fundamentals/typography/combining-text-presets-1

**Reply by AnaMontes11** [reply]
Los nombres los añadí a modo de explicar la idea. Iría con agnósticos precisamente por lo que comentas ;)

**Comment by aweell** [comment]
Añado por aquí una propuesta para el tema de los títulos y como esto podría resolver varios de los problemas que han ido apareciendo.

* El caso en el que fuera necesario cambiar el nivel semántico del título de un hero, esta necesidad fue levantada por un equipo recientemente.
* El caso que sea necesario cambiar la jerarquía visual de un título aislado o potencialmente aquellos componentes que tienen un título como elemento interno (el caso anterior del hero o el caso de un header. Necesidad actual del proyecto de Vivo.

La solución podría pasar por lo siguiente:

* Tener un componente Title que permita:
   * Cambiar su jerarquía visual
   * Cambiar su jerarquía semántica

(Podría ser un componente o añadir mas componentes a los actuales, en vez de tener un Title1 y Title2, tener una escala del 1-5)

Estos niveles podrán ser utilizados internamente dentro de aquellos componentes que sea necesario. Al igual que en una card se puede elegir la jerarquía de los botones que se utilizan, poder elegir la jerarquía tanto visual como semántica del un título dentro de un componente.

Como añadido, en el caso de que la escala no solucione de una manera global las necesidades de las distintas OBs, valorar la posibilidad de que las propiedades de cada uno de los niveles jerárquicos del title sean tokenizables, para de esta manera cubrir todas las necesidades que puedan aparecer.
